File: 13/hs/2.hs
import Data.List (elemIndex)
import Data.Maybe (fromJust)
import System.IO
import Utils (compare, sort)

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let all = filter (not . null) $ lines contents
      markers = ["[[2]]", "[[6]]"]
      all' = all ++ markers
      ordered = sort (\a b -> Utils.compare a b == -1) all' -- Order - smallest first, largest last
      indices = map ((+ 1) . fromJust . (`elemIndex` ordered)) markers -- Get index of markers in sorted list (1-index based)
      key = product indices
  print ("Decoder Key: " ++ show key)
  hClose h

File: 13/js/2.js
const fs = require('fs');
const { compare } = require('./utils');

(function () {
  const contents = fs.readFileSync("../input.txt").toString().trim();
  const lists = contents.split("\r\n").map(x => x.trim()).filter(x => x.length > 0); // Extract lists
  const markers = ["[[2]]", "[[6]]"];
  lists.push(...markers); // Add markers
  const sorted = lists.sort((a, b) => compare(a, b) ? -1 : 0); // Sort
  const indices = markers.map(m => sorted.indexOf(m) + 1); // Get indices of markers
  const key = indices.reduce((a, b) => a * b, 1); // Calculate key by taking the product
  console.log("Decoder key: " + key);
})();